src/id_token_provider.erl

```diff
@@ -65,7 +66,7 @@ handle_cast(_Request, State) ->
   {noreply, State}.
 
 handle_info({refresh, Provider}, State) ->
-  #{exp_at := ExpAt} = refresh(Provider),
+  {ok, #{exp_at := ExpAt}} = refresh(Provider),
```

Comment:
Wouldn't this still crash the gen_server at startup if async revalidate is enabled and a provider is not working as expected?
Perhaps checking for an error and schedule a new timer?

Reply:
yes, it will, but I'm thinking to disable this completely in the app config anyway and didn't want to deal with that (I had some code for it already, but then thought that maybe you want it to crash in that case?! I don't know...)